src/bluesearch/database/topic.py

```diff
@@ -297,6 +297,88 @@ def get_topics_for_pmc_article(
     return journal_topics
 
 
+def get_topics_for_arxiv_articles(
+    arxiv_paths: Iterable[pathlib.Path] | Iterable[str], batch_size: int = 400
```

Comment:
Regarding the `batch_size` I would imagine the default of 400 is reasonable and does not lead to any problems, right? Since there is no way the CLI user could control it.

Reply:
In my experiments `400` seemed indeed a good number that allowed to retrieve large batches without having an error response from the server:
- in principle, the `arXiv` API can retrieve up to `1,000` different items per query
- but if you concatenate too many `arXiv id`s values in the query, then the URL becomes too long and the `arXiv` server returns error.

---

> there is no way the CLI user could control it

This is true, but I couldn't really think of a clean way to implement it.
- Adding a `batch_size` parameter to `argparse` would only make this parameter applicable to the case of `arxiv` source, while for any other sources it would be ignored.
- We could have a sort of `**kwargs` capture of extra parameters passed by command line and try to unpack it when calling the various methods, but this idea of capturing arbitrary parameters with `argparse` seems to be a bad design practice as it makes things not very transparent (see e.g. comments [here](parameter)).

What would you suggest?

Reply:
I think it is OK the way it is now.

One way of handling extra arguments would be https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_known_args

However, as you mentioned I would avoid doing if possible to avoid extra complexity.